BUILD.md

```diff
@@ -1,13 +1,48 @@
-## Building Pester
+# Building Pester
+
+Pester is written in Powershell and C#.  The Microsoft .NET 4.5.2 Framework is required to build the Pester binaries.
 
 Pester has a C# Solution which requires .Net Framework SDKs and Developer Packs in order to compile. The targetted frameworks can be found in `src\csharp\Pester\Pester.csproj`.
 
-### Required Software
+## Required Software
 
-#### Install .NET Core 3.1 SDK
+### Install .NET Core 3.1 SDK
 
 [Download Link](https://dotnet.microsoft.com/download/dotnet-core/3.1)
 
-#### .Net Framework 4.5 Developer Pack
+### .Net Framework 4.5 Developer Pack
 
 [Download Link](https://dotnet.microsoft.com/download/dotnet-framework/net452)
+<https://aka.ms/msbuild/developerpacks>
+
+## Running Tests
+
+In Powershell, run test.ps1.  This defines the inherited function InPesterModuleScope and some types required for the tests.
+
+Afterwards, each test can be run individually using Invoke-Pester.
+
+Test.ps1 and optionally -skipPTests to skip the .ts.ps1 files.
+
+## test.ps1
+
+test.ps1 can be run with the following parameters
+
+TODO document parameters, tests executed in pipelines.
```

Comment:
Need some additional help for the parameters used in test.ps1

Reply:
param (
    # forces P Tests  to fail when I leave `dt` in the tests. dt only runs the specified test, so leaving it in code would run only one test from the file on the server. Also enable exit with 1 if tests don't pass
    [switch]$CI,
    # skip the tests written using the P module, those are unit tests for the runtime, and acceptance tests for whole Pester
    [switch]$SkipPTests,
    # do not build the underlying csharp part, used in CI where we already built it from clean
    [switch]$NoBuild,
    # pass the file to run Pester (not P) tests from
    [string[]] $File
)

Reply:
Ah you already described it.